src/v2/stack_aggregator.py

```diff
@@ -309,13 +312,14 @@ def initiate_unknown_package_ingestion(self):
             return
 
         ecosystem = self._normalized_packages.ecosystem
+        pkg_list = self.get_all_unknown_packages()
         try:
-            for dep in self.get_all_unknown_packages():
-                server_create_analysis(ecosystem, dep.name, dep.version, api_flow=True,
-                                       force=False, force_graph_sync=True)
-        except Exception as e:  # pylint:disable=W0703,C0103
-            logger.error('Ingestion failed for {%s, %s, %s}',
-                         ecosystem, dep.name, dep.version)
+            unknown_pkgs = set(map(lambda pkg: _PACKAGE(package=pkg.name,
+                                                        version=pkg.version), pkg_list))
+            unknown_package_flow(ecosystem, unknown_pkgs)
+            logger.debug('Unknown ingestion executed for %s packages in %s ecosystem',
+                         len(pkg_list), ecosystem)
+        except Exception as e:
             logger.error(e)
```

Comment:
Exception should be handled only for the statement which would potentially raise the exception.
```suggestion
            unknown_pkgs = set(map(lambda pkg: _PACKAGE(package=pkg.name,
                                                        version=pkg.version), pkg_list))
       try:
            unknown_package_flow(ecosystem, unknown_pkgs)
        except Exception as e:
            logger.error('Unknown ingestion failed with %s', e)
        else:
            logger.debug('Unknown ingestion executed for %s packages in %s ecosystem',
                         len(pkg_list), ecosystem)
```